File: npm/templates/base/scripts/configureProvider.ts
import { readFileSync, renameSync, writeFileSync } from "node:fs";

import { ProviderSigner } from "../src/providerSigning.js";

const MANIFEST = "capability.json";

const signer = new ProviderSigner();
const manifest = JSON.parse(readFileSync(MANIFEST, "utf8")) as Record<string, unknown>;
manifest.provider_pubkey = signer.publicKeyBase64;
const temporary = `${MANIFEST}.tmp`;
writeFileSync(temporary, `${JSON.stringify(manifest, null, 2)}\n`, "utf8");
renameSync(temporary, MANIFEST);
console.log(`provider identity ready: ${signer.publicKeyBase64}`);

File: src/create_aimarket_agent/templates/base/configure_provider.py
from __future__ import annotations

import json
from pathlib import Path

from provider_signing import ProviderSigner


def main() -> int:
    signer = ProviderSigner()
    path = Path("capability.json")
    manifest = json.loads(path.read_text(encoding="utf-8"))
    manifest["provider_pubkey"] = signer.public_key_b64
    temporary = path.with_suffix(".json.tmp")
    temporary.write_text(json.dumps(manifest, ensure_ascii=False, indent=2) + "\n", encoding="utf-8")
    temporary.replace(path)
    print(f"provider identity ready: {signer.public_key_b64}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
